source/common/http/http1/codec_impl.h

```diff
@@ -317,6 +317,8 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
 
   static http_parser_settings settings_;
 
+  bool dispatching_ : 1;
```

Comment:
Nit: prefer default initialize scalars with `{}` (this might be necessary to allow the suggested `ASSERT`s to work.

Reply:
This one was initialized like this in the H/2 codec, however in a new connection it is initialized as false... it makes more sense for me to initialize false, so I did `{false}`.